core/parentchain/indirect-calls-executor/src/executor.rs

```diff
@@ -189,9 +190,17 @@ impl<
 		let call = self.node_meta_data_provider.get_from_metadata(|meta_data| {
 			meta_data.confirm_processed_parentchain_block_call_indexes()
 		})??;
-
 		let root: H256 = merkle_root::<Keccak256, _>(extrinsics);
-		Ok(OpaqueCall::from_tuple(&(call, block_hash, block_number, root)))
+		let mrenclave = self.stf_enclave_signer.ocall_api.get_mr
+		let shard = self.top_pool_author.get_shards().get(0).unwrap_or(),
+		trace!("prepared confirm_processed_parentchain_block() call for block {:?} with index {:?} and merkle root {}", block_number, call, root);
+		Ok(OpaqueCall::from_tuple(&(
+			call,
+			shard,
```

Comment:
@clangenb we need to get the shard into this extrinsic, but it ain't obvious how to get it. get_shards from top_pool_author yields an empty vec

Reply:
The `top_pool_author` has access to the [state_facace](https://github.com/integritee-network/worker/blob/d1b8cbefe9eb09228754063060a50d4c244727f4/core-primitives/top-pool-author/src/author.rs#L81), which in turn has access to all shards: https://github.com/integritee-network/worker/blob/d1b8cbefe9eb09228754063060a50d4c244727f4/core-primitives/stf-state-handler/src/query_shard_state.rs#L31

So we need to expose this method on the level of the top-pool-author.

Reply:
ok. seems to work